### PR TITLE
Reduce public aggregator API to match latest specification.

### PR DESCRIPTION
**Pull request by jsuereth**
*BREAKING SDK API CHANGE*

## Specification
See [aggregation specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#aggregation).

HIghlights:
- Aggregation is only exposed as name + configuration on Views.
- A limited set of aggregations are defined for the first release
- There are no custom aggregations in the first version of the SDK.

## Changes in this PR
- Move existing aggregators into an internal package with appropriate caveats.  This allows us to adapt aggregator interface over time.
- Expose "Aggregation" public API that aligns with current SDK specification around view aggregators.
- Add "none" aggregation which disables metric generation.
- Remove `findView` from view registry
- Remove explicit histogram bucket aggregation configuration from JMH tests now that histogram is the default.

This is intended to be done in such a way that we can refactor/update Aggregator code in non-breaking ways going forward.

Note: There is an AggregationExtension class that is meant to be NON-specified API for metrics.   Given feedback, I'll be adapting this to move those extensions to an sdk-extension class.

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/AggregationExtension.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.internal.aggregator.AggregatorFactory;
+import java.util.function.Function;
+
+/**
+ * Extensions for configuring aggregation on Views.
+ *
+ * <p>This class provides aggregation not included in the OpenTelemetry metrics SDK specification.
+ */
+public final class AggregationExtension {
```
[the remark this answers is not in the document]
Comment (jsuereth) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/AggregationExtension.java]: done.

**Comment by anuraaga** [comment]
Linux failure is related to #3565

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  public abstract AggregatorFactory config(InstrumentDescriptor instrument);
+
+  /** The None Aggregation will ignore/drop all Instrument Measurements. */
+  public static Aggregation none() {
+    return NONE;
+  }
+
+  /** The default aggregation for an instrument will be chosen. */
+  public static Aggregation defaulAggregation() {
```
Comment (anuraaga) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java:75]:
```suggestion
  public static Aggregation defaultAggregation() {
```

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    return DEFAULT;
+  }
+
+  /** Instrument measurements will be combined into a metric Sum. */
+  public static Aggregation sum(AggregationTemporality temporality) {
+    return Aggregation.make("sum", unused -> AggregatorFactory.sum(temporality));
+  }
+
+  /** Instrument meaasurements will be combined into a metric Sum. */
```
Comment (anuraaga) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java:84]:
```suggestion
  /** Instrument measurements will be combined into a metric Sum. */
```

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  public static Aggregation explictBucketHistogram(AggregationTemporality temporality) {
+    return explictBucketHistogram(temporality, DEFAULT_HISTOGRAM_BUCKET_BOUNDARIES);
+  }
+
+  /**
+   * Aggregates measurments into an explicit bucket histogram.
+   *
+   * @param temporality Whether to report DELTA or CUMULATIVE metrics.
+   * @param bucketBoundaries A list of (inlcusive) upper bounds for the histogram. Should be in
```
Comment (anuraaga) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java:114]:
```suggestion
   * @param bucketBoundaries A list of (inclusive) upper bounds for the histogram. Should be in
```

@jsuereth Your spell-checker keeps on turning itself off? :P

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/ViewBuilder.java [hunk 5]
```diff
@@ -5,14 +5,13 @@
...
 
-import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
 import io.opentelemetry.sdk.metrics.processor.LabelsProcessorFactory;
 
 /** Builder of metric {@link View}s. */
 public final class ViewBuilder {
   private String name = null;
   private String description = null;
-  private AggregatorFactory aggregatorFactory;
+  private Aggregation aggregation = Aggregation.defaulAggregation();
```
Comment (anuraaga) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/ViewBuilder.java:14]:
```suggestion
  private Aggregation aggregation = Aggregation.defaultAggregation();
```

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java [hunk 6]
```diff
@@ -211,32 +212,33 @@ void collectAllSyncInstruments_OverwriteTemporality() {
...
   }
 
   @Test
   @SuppressWarnings("unchecked")
-  void collectAllSyncInstruments_DeltaCount() {
+  void collectAllSyncInstruments_DeltaHistogram() {
     registerViewForAllTypes(
-        sdkMeterProviderBuilder, AggregatorFactory.count(AggregationTemporality.DELTA));
+        sdkMeterProviderBuilder,
+        Aggregation.explictBucketHistogram(AggregationTemporality.DELTA, Collections.emptyList()));
```
Comment (anuraaga) [inline sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java:223]: `Aggregation.` sounds nicer than `AggregatorFactory.` good improvement

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java (hunk 4 above)
Comment (jsuereth) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java:114]: looks like it is broken after the latest update, let me figure out how I messed that up.

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java (hunk 6 above)
Comment (jsuereth) [reply sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java:223]: thanks!

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java [hunk 7]
```diff
@@ -554,9 +554,11 @@ void viewSdk_AllowMulitpleViewsPerAsynchronousInstrument() {
 
   @Test
   @SuppressWarnings("unchecked")
-  void collectAllAsyncInstruments_CumulativeCount() {
```
Comment (jkwatson) [inline sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java:557]: did we intentionally lose the tests for counts?

Comment (jsuereth) [reply sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterProviderTest.java:557]: count aggregator isn't exposed publicly.  Instead using a histogram to do counts, yes.

**Review by jkwatson — APPROVED** [review]

**Comment by jkwatson** [comment]
this needs a rebase to pick up the collector image changes.

**Comment by jsuereth** [comment]
Rebased.  @anuraaga Let me know if you have additional comments. I'd like to do an extension to this for multi-exporter to vet the current PR in the spec repository, but it'll involve a LOT of refactoring in the internal directory.  I'd like to base it on this PR.

**Comment by jack-berg** [comment]
FYI, I noticed that this PR removes the ability to configure summary aggregations, which I recognize that this is inline with the spec. I've opened an [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1902) to discuss the lack of summary support. 

I'm not trying to hold up this PR.

**Review by anuraaga — APPROVED** [review]

**Comment by jsuereth** [comment]
@jack-berg Summary metrics are considered deprecated at this point.  You should be able to extract summary style metrics from histogram aggregation.   If you have time would be happy to have a conversion available as a utillity

**Pull request merged**